stats/stats/src/charts/db_interaction/read/mod.rs

```diff
@@ -28,19 +29,29 @@ pub enum ReadError {
     IntervalTooLarge(u32),
 }
 
-pub struct QueryAllBlockTimestampRange;
+pub struct QueryFullIndexerTimestampRange;
 
-impl RemoteQueryBehaviour for QueryAllBlockTimestampRange {
+impl RemoteQueryBehaviour for QueryFullIndexerTimestampRange {
     type Output = Range<DateTime<Utc>>;
 
     async fn query_data(
         cx: &UpdateContext<'_>,
         _range: UniversalRange<DateTime<Utc>>,
     ) -> Result<Self::Output, ChartError> {
-        let start_timestamp = get_min_date_blockscout(cx.indexer_db)
-            .await
-            .map_err(ChartError::IndexerDB)?
-            .and_utc();
+        let min_date = get_min_date(cx.indexer_db, cx.is_multichain_mode).await;
+
+        let start_timestamp = min_date.map_err(ChartError::IndexerDB)?.and_utc();
         Ok(start_timestamp..cx.time)
     }
 }
+
+pub async fn get_min_date(
+    indexer_db: &DatabaseConnection,
+    is_multichain: bool,
+) -> Result<NaiveDateTime, DbErr> {
+    if is_multichain {
+        get_min_date_multichain(indexer_db).await
```

Comment:
I like the idea, but I worry that it might lead to some errors.
`data_source_query_range_to_db_statement_range` replaces one unbounded side with the range returned by (now) `QueryFullIndexerTimestampRange`. if there are some interop transfers (for example), that are older than earliest imported counter (sounds reasonable), then the earliest date will be incorrect

for main blockscout it's the earliest block because no transfers/txns/blocks can be earlier than the first block

Reply:
Yes, I understood it when implementing this routine. But since the stats service cannot recover data for that period (when no counter data was imported), there is no reason to include this interval in the full time range. In other words, this time range reflects the period that is 'complete' for statistics. 

By the way, which possible errors did you mean?

Reply:
In current state yes.

However, if we add a chart like `newInteropTransfers` or similar that will show number of transfers for each day, the behaviour won't be correct.
Let's say we've indexed interop messages from May 1st, but also we've started the multichain indexer on May 15th. 
The counters would've been imported from the 15th as well.
But if a user were to query `newInteropTransfers` until May 31st (with no lower bound), the stats service would determine that only data from May 15th until May 31st is possible, therefore it won't return points for 1-14.

Reply:
Agree with that. Unfortunately, the `multichain-aggregator` database doesn’t have direct timestamp data that reflects the lower boundary of the time range (unlike the Blockscout instance database, where we can use the first block’s timestamp). Currently, only the `global_counters_imported` and `interop_messages` tables contain relevant data. Therefore, I’ve updated this routine to return the minimum timestamp from these tables. https://github.com/blockscout/blockscout-rs/pull/1485/commits/cbd6abad6cd41ee18836ee05f0dbeeaed26cac64